frontend/spec/features/search_spec.rb

```diff
@@ -0,0 +1,377 @@
+    find('.advanced-search-add-text-row').click
+
+    find('#v0').fill_in with: @keywords[0]
+    find('#v1').fill_in with: @keywords[4]
+    select 'Title', from: 'f1'
+
+    click_on 'Search'
+
+    find(:xpath, "//table//tr[contains(., '#{@accession_1.title}')]")
+
+    expect(page).to_not have_text @resource_1.title
+    expect(page).to_not have_text @digital_object_1.title
+    expect(page).to_not have_text @accession_2.title
+    expect(page).to_not have_text @resource_2.title
+    expect(page).to_not have_text @digital_object_2.title
+  end
+
+  it 'finds matches with two keyword ORed field queries' do
+    find('.navbar .search-switcher').click
+    find('.search-switcher-hide')
+
+    find('.advanced-search-add-row-dropdown').click
+    find('.advanced-search-add-text-row').click
+    find('#v0').fill_in with: @keywords[0]
+    find('#v1').fill_in with: @keywords[4]
+    select 'Title', from: 'f1'
+    select 'Or', from: 'op1'
+
+    click_on 'Search'
+
+    find(:xpath, "//table//tr[contains(., '#{@accession_1.title}')]")
+    find(:xpath, "//table//tr[contains(., '#{@resource_1.title}')]")
+    find(:xpath, "//table//tr[contains(., '#{@digital_object_1.title}')]")
+
+    expect(page).to_not have_text @accession_2.title
+    expect(page).to_not have_text @resource_2.title
+    expect(page).to_not have_text @digital_object_2.title
+  end
+
+  it 'finds matches with two keyword joined AND NOTed field queries' do
+    find('.navbar .search-switcher').click
+    find('.search-switcher-hide')
+    find('.advanced-search-add-row-dropdown').click
+    find('.advanced-search-add-text-row').click
+    find('#v0').fill_in with: @keywords[0]
+    find('#v1').fill_in with: @keywords[4]
+    select 'Title', from: 'f1'
+    select 'Not', from: 'op1'
+
+    click_on 'Search'
+
+    find(:xpath, "//table//tr[contains(., '#{@resource_1.title}')]")
+    find(:xpath, "//table//tr[contains(., '#{@digital_object_1.title}')]")
+
+    expect(page).to_not have_text @accession_1.title
+    expect(page).to_not have_text @accession_2.title
+    expect(page).to_not have_text @resource_2.title
+    expect(page).to_not have_text @digital_object_2.title
+  end
+
+  it 'clear resets the fields' do
+    find('.navbar .search-switcher').click
+    find('.search-switcher-hide')
+    find('.advanced-search-add-row-dropdown').click
+    find('.advanced-search-add-text-row').click
+
+    find('#v0').fill_in with: @keywords[0]
+    find('#v1').fill_in with: @keywords[4]
+    select 'Title', from: 'f1'
+    select 'Not', from: 'op1'
+
+    click_on 'Search'
+    click_on 'Clear'
+    expect(find('#v0').value).to eq('')
+  end
+
+  it 'allow adding of mulitple rows of the same type' do
+    find('.navbar .search-switcher').click
+    find('.search-switcher-hide')
+    find('.advanced-search-add-row-dropdown').click
+    find('.advanced-search-add-bool-row').click
+
+    find('.advanced-search-add-row-dropdown').click
+    find('.advanced-search-add-bool-row').click
+
```

Comment:
As success conditions, these might better be expressed as `expect(page).to have_selector('#v1')`